pkg/varcontext/varcontext.go

```diff
@@ -0,0 +1,72 @@
+// Copyright 2018 the Service Broker Project Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package varcontext
+
+import (
+	"fmt"
+
+	"github.com/spf13/cast"
+)
+
+type VarContext struct {
```

Comment:
Making `VarContext` an `ErrorCollector` has the weird side effect of exposing the `AddError` call. It's not a big deal, but it means that a varcontext caller can just add arbitrary errors to it. Instead, if `VarContext` had an internal field `ErrorCollector`, all it should have to do is implement `Error()` by calling into the `ErrorCollector`. That would stop the accidental exposure and should cover all needs (assuming HasError calls have also been replaced, see other comments)

Reply:
Fixed